runtime/core/frontend/fbank.h

```diff
@@ -125,26 +219,50 @@ class Fbank {
     (*data)[0] -= coeff * (*data)[0];
   }
 
-  // Apply povey window on data in place
-  void Povey(std::vector<float>* data) const {
-    CHECK_GE(data->size(), povey_window_.size());
-    for (size_t i = 0; i < povey_window_.size(); ++i) {
-      (*data)[i] *= povey_window_[i];
+  // Apply window on data in place
+  void ApplyWindow(std::vector<float>* data) const {
+    CHECK_GE(data->size(), window_.size());
+    for (size_t i = 0; i < window_.size(); ++i) {
+      (*data)[i] *= window_[i];
+    }
+  }
+
+  void WhisperNorm(std::vector<std::vector<float>>* feat,
+                   float max_mel_engery) {
+    int num_frames = feat->size();
+    for (int i = 0; i < num_frames; ++i) {
+      for (int j = 0; j < num_bins_; ++j) {
+        float energy = (*feat)[i][j];
+        if (energy < max_mel_engery - 8) energy = max_mel_engery - 8;
+        energy = (energy + 4.0) / 4.0;
+        (*feat)[i][j] = energy;
+      }
     }
   }
 
   // Compute fbank feat, return num frames
   int Compute(const std::vector<float>& wave,
               std::vector<std::vector<float>>* feat) {
     int num_samples = wave.size();
+
     if (num_samples < frame_length_) return 0;
     int num_frames = 1 + ((num_samples - frame_length_) / frame_shift_);
     feat->resize(num_frames);
     std::vector<float> fft_real(fft_points_, 0), fft_img(fft_points_, 0);
     std::vector<float> power(fft_points_ / 2);
+
+    float max_mel_engery = std::numeric_limits<float>::min();
+
     for (int i = 0; i < num_frames; ++i) {
       std::vector<float> data(wave.data() + i * frame_shift_,
                               wave.data() + i * frame_shift_ + frame_length_);
+
+      if (scaled_float_as_input_) {
+        for (int j = 0; j < frame_length_; ++j) {
+          data[j] = data[j] / S16_TO_FLOAT_SCALE;
```

Comment:
data feed into this pipeline is int but converted to float without scaling, whisper training code load this as float between -1 to 1